app/workers/end_of_cycle/cancel_unsubmitted_applications_worker.rb

```diff
@@ -0,0 +1,36 @@
+module EndOfCycle
+  class CancelUnsubmittedApplicationsWorker
+    include Sidekiq::Worker
+
+    BATCH_SIZE = 200
+
+    def perform(force: false)
+      return unless CycleTimetable.cancel_unsubmitted_applications? || force
+
+      BatchDelivery.new(relation:, stagger_over: 2.hours, batch_size: BATCH_SIZE).each do |batch_time, applications|
+        CancelUnsubmittedApplicationsSecondaryWorker.perform_at(batch_time, applications.pluck(:id))
+      end
+    end
+
+    def relation
+      ApplicationForm
+        .current_cycle
+        .joins(:application_choices).where('application_choices.status': 'unsubmitted')
+        .distinct
+    end
+  end
+
+  class CancelUnsubmittedApplicationsSecondaryWorker
+    include Sidekiq::Worker
+
+    def perform(application_form_ids)
+      application_form_ids.each do |application_form_id|
+        ApplicationForm.find(application_form_id).application_choices.unsubmitted.each do |application_choice|
```

Comment:
Can we change to not loop through each application_form_ids? That would be N+1 query? There would be 400 queries per job (Application form and application choices query) + 1 update. I wonder why Bullet didn't raise anything. 🤔